src/hotspot/cpu/x86/templateTable_x86.cpp

```diff
@@ -2721,13 +2721,84 @@ void TemplateTable::load_field_cp_cache_entry(Register obj,
   }
 }
 
+void TemplateTable::load_invokedynamic_entry(Register method) {
+  // setup registers
+  const Register appendix = rax;
+  const Register cache = rcx;
+  const Register index = rdx;
+  assert_different_registers(method, appendix, cache, index);
+
+  __ save_bcp();
+
+  Label resolved;
+
+  __ load_resolved_indy_entry(cache, index);
+  __ movptr(method, Address(cache, in_bytes(ResolvedIndyEntry::method_offset())));
+
+  // Compare the method to zero
+  __ testptr(method, method);
+  __ jcc(Assembler::notZero, resolved);
+
+  Bytecodes::Code code = bytecode();
+
+  // Call to the interpreter runtime to resolve invokedynamic
+  address entry = CAST_FROM_FN_PTR(address, InterpreterRuntime::resolve_from_cache);
+  __ movl(method, code); // this is essentially Bytecodes::_invokedynamic
+  __ call_VM(noreg, entry, method); // Example uses temp = rbx. In this case rbx is method
+  // Update registers with resolved info
+  __ load_resolved_indy_entry(cache, index);
+  __ movptr(method, Address(cache, in_bytes(ResolvedIndyEntry::method_offset())));
+
+#ifdef ASSERT
+  __ testptr(method, method);
+  __ jcc(Assembler::notZero, resolved);
+  __ stop("Should be resolved by now");
+#endif // ASSERT
+  __ bind(resolved);
+
+  Label L_no_push;
+  // Check if there is an appendix
+  __ load_unsigned_byte(index, Address(cache, in_bytes(ResolvedIndyEntry::flags_offset())));
+  __ testl(index, (1 << ResolvedIndyEntry::has_appendix_shift));
+  __ jcc(Assembler::zero, L_no_push);
+
+  // Get appendix
+  __ load_unsigned_short(index, Address(cache, in_bytes(ResolvedIndyEntry::resolved_references_index_offset())));
+  // Push the appendix as a trailing parameter
+  // since the parameter_size includes it.
+  __ push(rbx);
+  __ mov(rbx, index);
```

Comment:
Why is the index (rdx) copied to rbx instead of using the index (rdx) register directly to call load_resolved_reference_at_index() ? The method doesn't modify the content of the register.